Add aws detailed monitoring toggle

pcn opened the pull request:
### What does this PR do?

Adds cloud functionality for getting and setting the ec2 instance detailed monitoring property.

### What issues does this PR fix or reference?

The aws cloud cli and api lacks the ability to enable and disable detailed monitoring (cloudwatch metrics on the instance).  This provides the ability to do this.

### New Behavior

```
(salt)root@saltmaster-3-stagevpc:~/salt# salt-cloud -y -a disable_detailed_monitoring zksvcs-1-stagevpc
librato-ec2-private-ips:
    ----------
    ec2:
        ----------
        zksvcs-1-stagevpc:
            ----------
            state:
                disabling
(salt)root@saltmaster-3-stagevpc:~/salt# salt-cloud -y -a show_detailed_monitoring zksvcs-1-stagevpc
librato-ec2-private-ips:
    ----------
    ec2:
        ----------
        zksvcs-1-stagevpc:
            ----------
            state:
                disabled
(salt)root@saltmaster-3-stagevpc:~/salt# salt-cloud -y -a enable_detailed_monitoring zksvcs-1-stagevpc
librato-ec2-private-ips:
    ----------
    ec2:
        ----------
        zksvcs-1-stagevpc:
            ----------
            state:
                pending
(salt)root@saltmaster-3-stagevpc:~/salt# salt-cloud -y -a show_detailed_monitoring zksvcs-1-stagevpc
librato-ec2-private-ips:
    ----------
    ec2:
        ----------
        zksvcs-1-stagevpc:
            ----------
            state:
                enabled
```

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

pcn committed:
Add calls for detailed monitoring

pcn committed:
Fix result creation

pcn committed:
Now I understand how this is supposed to work.

pcn committed:
Fix camel-mis-case

salt-jenkins commented:
@pcn, thanks for your PR! By analyzing the history of the files in this pull request, we identified @techhat, @eliasp and @s0undt3ch to be potential reviewers.

cachedout reviewed (changes requested):
This failed some lint checks:

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/10508/violations/file/salt/cloud/clouds/ec2.py/

Could you please address those two issues? Thanks.

pcn committed:
Committing an empty line to re-run CI

pcn committed:
Fix 1 line and 3 lines between functions

Fixes pylint complaints from jenkins:

17:29:08 salt/cloud/clouds/ec2.py:3740: [E8302(expected-2-blank-lines-found-0), ] PEP8 E302: expected 2 blank lines, found 0
17:29:08 salt/cloud/clouds/ec2.py:3768: [E8303(too-many-blank-lines-3), ] PEP8 E303: too many blank lines (3)

pcn commented:
@cachedout those pylint spacing errors are clear.
